code/modules/hydroponics/spreading/spreading_response.dm

```diff
@@ -25,13 +26,99 @@
 		spawn(1)
 			entangle(M)
 
+
+/*************************
+	Attack procs
+**************************/
 /obj/effect/plant/attack_hand(var/mob/user)
 	manual_unbuckle(user)
 
 /obj/effect/plant/attack_generic(var/mob/user)
 	if(istype(user))
 		manual_unbuckle(user)
 
+/obj/effect/plant/attackby(var/obj/item/weapon/W, var/mob/user)
+
+	user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN*1.5)
+	plant_controller.add_plant(src)
+	if(istype(W, /obj/item/weapon/reagent_containers/syringe))
+		return
+
+
+	if(istype(W, /obj/item/weapon/tool/wirecutters) || istype(W, /obj/item/weapon/tool/scalpel))
+		if(sampled)
+			user << SPAN_WARNING("\The [src] has already been sampled recently.")
+			return
+		if(!is_mature())
+			user << SPAN_WARNING("\The [src] is not mature enough to yield a sample yet.")
+			return
+		if(!seed)
+			user << SPAN_WARNING("There is nothing to take a sample from.")
+			return
+		if(sampled)
+			user << SPAN_DANGER("You cannot take another sample from \the [src].")
+			return
+		if(prob(70))
+			sampled = 1
+		seed.harvest(user,0,1)
+		health -= (rand(3,5)*5)
+		sampled = 1
+		return
+	else
+		//Gardening tools can cut down vines quickly
+		var/tool_type = W.get_tool_type(user, list(QUALITY_SHOVELING, QUALITY_CUTTING))
```

Comment:
Welding quality would also make sense here.

Reply:
the more qualities i add, the more likely that'll happen